pkg/userdata/helper/kubelet.go

```diff
@@ -0,0 +1,109 @@
+package helper
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"text/template"
+)
+
+const (
+	kubeletFlagsTpl = `--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf \
+--kubeconfig=/etc/kubernetes/kubelet.conf \
+--pod-manifest-path=/etc/kubernetes/manifests \
+--allow-privileged=true \
+--network-plugin=cni \
+--cni-conf-dir=/etc/cni/net.d \
+--cni-bin-dir=/opt/cni/bin \
+--authorization-mode=Webhook \
+--client-ca-file=/etc/kubernetes/pki/ca.crt \
+{{- if semverCompare "<1.12.0-0" .KubeletVersion }}
+--cadvisor-port=0 \
+{{- end }}
+--rotate-certificates=true \
+--cert-dir=/etc/kubernetes/pki \
+--authentication-token-webhook=true \
+{{- if .CloudProvider }}
+--cloud-provider={{ .CloudProvider }} \
+--cloud-config=/etc/kubernetes/cloud-config \
+{{- end }}
+--hostname-override={{ .Hostname }} \
+--read-only-port=0 \
+--protect-kernel-defaults=true \
```

Comment:
Kubeadm does not set this, do we really want it?

Reply:
We intentionally added this flag on a customer request after a fair amount of research.
cc @kron4eg